nequip/train/schedulefree.py

```diff
@@ -0,0 +1,58 @@
+import torch
+from .lightning import NequIPLightningModule
+from schedulefree import AdamWScheduleFree
+from nequip.data import AtomicDataDict
+from typing import Optional, Dict
+
+
+class ScheduleFreeLightningModule(NequIPLightningModule):
+    """
+    NequIP LightningModule using Facebook's Schedule-Free optimizer.
+
+    This module wraps the model's optimizer in AdamWScheduleFree to enable
+    schedule-free training. See: https://github.com/facebookresearch/schedule_free
+
+    Args:
+        optimizer (Dict): Dictionary of keyword arguments compatible with
+            AdamWScheduleFree. See: https://github.com/facebookresearch/schedule_free/blob/main/schedulefree/adamw_schedulefree.py
+    """
+
+    def __init__(
+        self,
+        optimizer: Dict,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.automatic_optimization = True
+        self.optimizer_params = optimizer
+        self._sf_optimizer = None
+
+    def configure_optimizers(self):
```

Comment:
I don't think we have to override this method. Instead I think we should just do checks at `__init__` to make sure `optimizer` arg conforms to the set of `ScheduleFree` optimizers, i.e. check that their `_target_` is in `["SGDScheduleFree", ....]`. Then we can rely on the existing logic to instantiate the optimizers correctly, for reuse of the base class's `configure_optimizers` method (which is also safer) https://github.com/mir-group/nequip/blob/60eae375988b07a35b0840eff7db42de076c9a73/nequip/train/lightning.py#L191